crates/languages/src/go.rs

```diff
@@ -111,11 +112,17 @@ impl super::LspAdapter for GoLspAdapter {
         container_dir: PathBuf,
         delegate: &dyn LspAdapterDelegate,
     ) -> Result<LanguageServerBinary> {
+        let go = delegate.which("go".as_ref()).await.unwrap_or("go".into());
+        let go_version_output = util::command::new_smol_command(&go)
+            .args(["version"])
+            .output()
+            .await?;
```

Comment:
Can you add a `.context` here to make it clear that "failed to run `go version`" or something?